### PR TITLE
refactor: cleanup the code for CORS handling

### PR DESCRIPTION
**Pull request by harsimranmaan**
## Related issue

#1754 

## Proposed changes

This change was originally intended to fix the issue described in the bug. However, I am not able to repro it as shown in the newly added test.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [N/A] I have added or changed [the documentation](docs/docs).

## Further comments

I removed many extra loops for validating the origin against the CORS allowlist. The viper config was shared across test cases causing dependency on the order of the test cases. It also hid some assumptions about the setup.

### CODE REVIEW

File: driver/cors_test.go [hunk 1]
```diff
@@ -168,9 +178,21 @@ func TestOAuth2AwareCORSMiddleware(t *testing.T) {
...
+			d: "should accept when basic auth client exists and origin is set at the client as well as the server",
+			prep: func() {
+				viper.Set("serve.public.cors.enabled", true)
+				viper.Set("serve.public.cors.allowed_origins", []string{"http://**.example.com"})
+				r.ClientManager().CreateClient(context.Background(), &client.Client{ID: "foo8", Secret: "bar", AllowedCORSOrigins: []string{"http://myapp.example.biz"}})
+			},
+			code:         http.StatusNotImplemented,
+			header:       http.Header{"Origin": {"http://client-app.example.com"}, "Authorization": {"Basic Zm9vLTU6YmFy"}},
+			expectHeader: http.Header{"Access-Control-Allow-Credentials": []string{"true"}, "Access-Control-Allow-Origin": []string{"http://client-app.example.com"}, "Access-Control-Expose-Headers": []string{"Content-Type"}, "Vary": []string{"Origin"}},
+		},
```
Comment (aeneasr) [inline driver/cors_test.go:191]: I think the bug that was reported would be reproduced like this (checking against the origin domain, not the server one):

```

 		{
 			d: "should accept when basic auth client exists and origin is set at the client as well as the server",
 			prep: func() {
 				viper.Set("serve.public.cors.enabled", true)
 				viper.Set("serve.public.cors.allowed_origins", []string{"http://**.example.com"})
 				r.ClientManager().CreateClient(context.Background(), &client.Client{ID: "foo8", Secret: "bar", AllowedCORSOrigins: []string{"http://myapp.example.biz"}})
 			},
 			code:         http.StatusNotImplemented,
 			header:       http.Header{"Origin": {"http://myapp.example.biz"}, "Authorization": {"Basic Zm9vLTU6YmFy"}},
 			expectHeader: http.Header{"Access-Control-Allow-Credentials": []string{"true"}, "Access-Control-Allow-Origin": []string{"http://client-app.example.com"}, "Access-Control-Expose-Headers": []string{"Content-Type"}, "Vary": []string{"Origin"}},
 		},

```

Comment (harsimranmaan) [reply driver/cors_test.go:191]: Lemme give it a try. I exepect the Allow-Origin to be set to `http://myapp.example.biz`.

Comment (harsimranmaan) [reply driver/cors_test.go:202]: @aeneasr I could not repro the defect. Added the new test-case. Also refactored auth headers for readability.

Comment (aeneasr) [reply driver/cors_test.go:202]: Ok, in that case I don't think there's any more to do! And you are correct, the expected value was a copy/paste mistake on my end!

**Comment by aeneasr** [comment]
Awesome 🎉 

Thank you for your contribution!

**Pull request merged**